### PR TITLE
=str small improvement in GraphInterpreter.currentInterpreter

### PR DESCRIPTION
**Pull request by rkuhn**
reduce the number of hash table lookups by a factor of three

### CODE REVIEW

**Comment by drewhk** [comment]
Sneaky. *Now* I see the purpose of that array!

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala [hunk 1]
```diff
@@ -518,7 +519,7 @@ private[stream] final class GraphInterpreter(
         eventsRemaining -= 1
       }
     } finally {
-      setCurrentInterpreter(previousInterpreter)
+      currentInterpreterHolder(0) = previousInterpreter
```
Comment (2m) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala]: After this change `setCurrentInterpreter`, `currentInterpreterOrNull` and `currentInterpreter` is not used anymore.

Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala]: Oh, right, the setter is indeed not used anymore, but I anticipate that we will need both getters in the immediate future—when running interpreters within interpreters for fused stream-of-stream stages.

**Comment by ktoso** [comment]
LGTM, very nice trick :-) Good to have Bjorn around to pitch those.

**Pull request merged**